.github/workflows/validate_docker_images.yml

```diff
@@ -0,0 +1,36 @@
+name: Validate Docker Images (with Matrix Generation)
+on:
+  workflow_dispatch:
+    inputs:
+      channel:
+        description: 'PyTorch channel to use (nightly, test, release, all)'
+        required: true
+        type: choice
+        default: 'all'
+        options:
+          - 'nightly'
+          - 'test'
+          - 'release'
+          - 'all'
+  push:
+    branches:
+    - '*'
+jobs:
+  generate-matrix:
+    uses: pytorch/test-infra/.github/workflows/generate_docker_release_matrix.yml@main
+    with:
+      channel: ${{ inputs.channel }}
+    secrets: inherit
+  run-tests:
+    needs: generate-matrix
+    runs-on: ubuntu-latest
+    strategy:
+      matrix: ${{ fromJson(needs.generate-matrix.outputs.matrix) }}
+    container:
+      image: ghcr.io/pytorch/pytorch:2.2.2-cuda${{ matrix.cuda }}-cudnn${{ matrix.cudnn_version }}-${{ matrix.image_type }}
```

Comment:
Please change image to ``matrix.docker`` which should be now since this PR is merged: https://github.com/pytorch/test-infra/pull/5081

Reply:
Updated and workflow partially succeeded: https://github.com/juliagmt-google/builder/actions/runs/8635045506/job/23672552152

The failed workflow complained about not enough space:
` failed to register layer: write /opt/conda/lib/python3.10/test/support/__init__.py: no space left on device
  Warning: Docker pull failed with exit code 1, back off 3.699 seconds before retry.
  /usr/bin/docker --config /home/runner/work/_temp/.docker_ff254e09-5ee1-4d53-9f39-52c9c2a6a945 pull ghcr.io/pytorch/pytorch-nightly:2.4.0.dev20240410-cuda11.8-cudnn8-devel`